src/app/api-server/src/config.rs

```diff
@@ -94,3 +107,75 @@ pub struct RepoCachingConfig {
     // Stores metadata blocks in a local directory to avoid many tiny S3 requests
     pub metadata_local_fs_cache_path: Option<PathBuf>,
 }
+
+/////////////////////////////////////////////////////////////////////////////////////////
+// UrlConfig
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct UrlConfig {
+    #[serde(deserialize_with = "parse_repo_url")]
+    pub base_url_platform: Url,
+    #[serde(deserialize_with = "parse_repo_url")]
```

Comment:
I think this could be avoided by enabling `serde` feature on `url` crate.

Reply:
Since we are using a function inside `parse_repo_url()` with our `value_parse_repo_url()`, it was not possible to use the proposed functionality